modules/core/src/main/java/org/apache/ignite/internal/processors/query/running/SqlPlanHistoryTracker.java

```diff
@@ -47,7 +42,7 @@ public SqlPlanHistoryTracker(int historySize) {
      * @param engine SQL engine.
      */
     public void addPlan(String plan, String qry, String schema, boolean loc, String engine) {
-        if (historySize <= 0)
+        if (sqlPlanHistory.getClass().equals(Collections.emptyMap().getClass()))
```

Comment:
We can't use `sqlPlanHistory == Collections.emptyMap()` because

_Operator '==' cannot be applied to 'java.util.Map<org.apache.ignite.internal.processors.query.running.SqlPlan,java.lang.Long>', 'java.util.Map<java.lang.Object,java.lang.Object>'_

We also can use `sqlPlanHistory.equals(Collections.emptyMap())` because `AbstractMap#equals` identifies two map as equal if they both are empty. So `sqlPlanHistory.equals(Collections.emptyMap())` will return `TRUE` for a newly created `sqlPlanHistory` with `size=0`, and the method won't work correctly.

If however we compare classes of `sqlPlanHistory` and `Collections.emptyMap()`, then the method works as we intended and `sqlPlanHistory.getClass().equals(Collections.emptyMap().getClass())` doesn't return `TRUE` for a newly created `sqlPlanHistory`.

Reply:
Something like
```
        Map<SqlPlan, Long> emptyMap = Collections.emptyMap(); 

        if (sqlPlanHistory == emptyMap)
            return;
```
?

Reply:
Yes, this options works. Change applied.